.github/workflows/dev.yml

```diff
@@ -339,17 +365,26 @@ jobs:
       - id: "clippy"
         name: "run clippy"
         run: |
-          just debug_justfile="${{matrix.debug_justfile}}" profile=${{matrix.profile.name}} \
-            ${{matrix.profile.sterile}} cargo clippy --all-targets --all-features -- -D warnings
+          if [ "${{matrix.profile.sterile}}" != "sterile" ]; then
+            # you can't really get away with --all-features here since that enables dpdk
+            cargo clippy --all-targets -- -D warnings
+          else
+            just debug_justfile="${{matrix.debug_justfile}}" profile=${{matrix.profile.name}} \
+                ${{matrix.profile.sterile}} cargo clippy --all-targets --all-features -- -D warnings
+          fi
 
       - id: "docs"
         name: "run rustdoc"
         run: |
-          RUSTDOCFLAGS="-D warnings" just \
-            debug_justfile="${{matrix.debug_justfile}}" \
-            profile=${{matrix.profile.name}} \
-            target=x86_64-unknown-linux-gnu \
-            ${{matrix.profile.sterile}} cargo doc --no-deps
+          if [ "${{matrix.profile.sterile}}" != "sterile" ]; then
+            cargo doc --no-deps
+          else
+            RUSTDOCFLAGS="-D warnings" just \
+                debug_justfile="${{matrix.debug_justfile}}" \
+                profile=${{matrix.profile.name}} \
+                target=x86_64-unknown-linux-gnu \
+                ${{matrix.profile.sterile}} cargo doc --no-deps
+          fi
```

Comment:
If you rebase, you should see an additional step just below this one, for running the doctests - You probably want to update that one, too.